nightfall-deployer/contracts/Challenges.sol

```diff
@@ -35,7 +35,9 @@ contract Challenges is Stateful, Key_Registry, Config {
   ) external {
     checkCommit(msg.data);
     // check if the block hash is correct and the block hash exists for the block and prior block
+    if (blockL2.blockNumberL2 != 0) {
```

Comment:
Do we need to check if a block that is passed in as `blockNumberL2 ==0` is indeed an identity block?

Reply:
Do you mean check whether it is indeed genesis block in L2? A block which passes `blockNumberL2 ==0` will be genesis block.  The identity block value is assigned to `priorBlock` and/or `priorPriorBlock` when  `blockNumberL2 ==0` .

Reply:
> A block which passes blockNumberL2 ==0 will be genesis block.

This is not necessarily true, a bad actor can submit a block as part of the input to this function where the `blockNumberL2 == 0` but the `leafCount > 0` which will incorrectly bypass this check right?

Reply:
@dwebchapey

Reply:
Added checks to confirm if priorBlock is `IDENTITY_BLOCK`

Reply:
@IlyasRidhuan Has this been resolved?